pkg/manager/setconfig.go

```diff
@@ -99,6 +101,10 @@ func (m *Manager) computeNewNetworkConfig(targetUpdates map[string]devicechanget
 		//FIXME this is a sequential job, not parallelized
 		version := deviceInfo[devicetype.ID(target)].Version
 		deviceType := deviceInfo[devicetype.ID(target)].Type
+		deviceType, version, err := mgr.CheckCacheForDevice(devicetype.ID(target), deviceType, version)
```

Comment:
this was already done at the `Set` level - line 112 of `set.go`. Also whatever you do to updates, you must do to **removes** also - i.e. the loop below

Reply:
I'd avoid doing this check there, duplicated cycles spent. move the tests in set_test.go and we shoudl be ok.